support/elasticsearch2_search.erl

```diff
@@ -442,26 +465,37 @@ map_query({text, Text}, Context) ->
         <<"title*^2">>
     ],
     Fields = z_notifier:foldr(#elasticsearch_fields{query = Text}, DefaultFields, Context),
+    DefOpCfg = m_config:get_value(mod_elasticsearch2, default_operator, Context),
+    DefaultOperator = case z_convert:to_upper(z_convert:to_binary(DefOpCfg)) of
+        <<"AND">> = DefOp -> DefOp;
+        <<"OR">> = DefOp -> DefOp;
+        _ -> <<"AND">>
+    end,
     Query = case z_convert:to_bool(m_config:get_value(mod_elasticsearch2, no_automatic_wildcard, Context)) of
         true ->
             #{
                 <<"simple_query_string">> => #{
                     <<"query">> => Text,
                     <<"fields">> => Fields,
-                    <<"default_operator">> => <<"OR">>,
+                    <<"default_operator">> => DefaultOperator,
                     <<"flags">> => <<"ALL">>
                 }
             };
         false ->
             {SearchText, Ops} = add_wildcards(Text),
+            SearchText1 = iolist_to_binary([
+                    $(, SearchText, $),
+                    " | ",
+                    $", Text, $"
+                ]),
             #{
                 <<"simple_query_string">> => #{
-                    <<"query">> => SearchText,
+                    <<"query">> => SearchText1,
                     <<"fields">> => Fields,
-                    <<"default_operator">> => <<"OR">>,
+                    <<"default_operator">> => DefaultOperator,
                     <<"flags">> => case Ops of
                         default -> <<"ALL">>;
-                        prefix -> <<"PREFIX|WHITESPACE">>
+                        prefix -> <<"PREFIX|WHITESPACE|OR|PRECEDENCE|PHRASE">>
```

Comment:
I probably should read more about this, at the moment I don't know what it does ;)

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html#supported-flags